### DISCUSSION TITLE
为什么xrepo会出现部分包无法搜索到的情况？

### DISCUSSION THREAD

**Discussion by free2king** [xmake-io/xmake#2191] [Q&A]
PS C:\Users\free> xrepo search zlib
The package names:
    zlib:
      -> chromium_zlib-2022.02.22: zlib from chromium (in xmake-repo)
PS C:\Users\free> xrepo search glfw
The package names:
PS C:\Users\free> xrepo search gli
The package names:

**Comment by waruqi** [comment]
进到本地仓库，执行 git log 看下，确认下有没有拉到最新仓库

```console
cd %localappdata%/.xmake/repositories/xmake-repo
git log
```

然后 git remote -v 看下，upstream url

**Reply by free2king** [reply]
commit a866090956c8d7375e71feafa27455697fb938b6 (HEAD -> master, origin/master, origin/dev, origin/HEAD)
Author: ruki <waruqi@gmail.com>
Date:   Fri Mar 18 00:15:39 2022 +0800

    Update xmake.lua

commit b5bba576899ba8d4834fe38083ec5a49c95110c5
Author: ruki <waruqi@gmail.com>
Date:   Fri Mar 18 00:11:19 2022 +0800

    Update xmake.lua

**Comment by free2king** [comment]
C:\Users\free\AppData\Local\.xmake\repositories\xmake-repo>git remote -v
origin  https://gitee.com/tboox/xmake-repo.git (fetch)
origin  https://gitee.com/tboox/xmake-repo.git (push)

**Reply by waruqi** [reply]
先执行 xrepo update-repo 后再看下 git log ，确保是否拉到最新更新，https://gitee.com/tboox/xmake-repo 上最近提交 15hour 以前

**Reply by waruqi** [reply]
如果已经确保最新，然后进入 xmake-repo/packages 找下，对应包的子目录是否存在

**Reply by free2king** [reply]
PS C:\Users\free> xrepo update-repo
updating repositories .. error: Updating a866090..f51e6fd
ox/xmake-repo
 * branch            master     -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
        packages/l/libcurl/xmake.lua
        packages/l/libsdl/xmake.lua
        packages/l/libwfkafka/xmake.lua
        packages/p/python/xmake.lua
        packages/q/qt5lib/xmake.lua
        packages/w/workflow/xmake.lua
        packages/z/zstd/xmake.lua
Please commit your changes or stash them before you merge.
Aborting

error: execv(xmake repo --update) failed(-1)

**Reply by free2king** [reply]
他报这错 .....

**Reply by waruqi** [reply]
本地 `C:\Users\free\AppData\Local\xmake\repositories\xmake-repo` 目录，每次都是 git pull 拉的，按理不会去改里面的文件。。但是你这里的问题是，里面很多的文件被改掉了，有可能被删了，有可能被篡改了，。。可以执行 `git status` 和 `git diff` 看下改动情况。。


反正，git 仓库被改，导致每次 update 去 pull 拉取失败，冲突了。。我不知道你这为什么会被改。但如果要修复，就执行下面的命令

```
git reset --hard HEAD
xrepo update-repo
```

**Reply by free2king** [reply]
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .github/
        .gitignore
        CONTRIBUTING.md
        LICENSE.md
        NOTICE.md
        README.md
        README_zh.md
        packages/
        updated


C:\Users\free\AppData\Local\.xmake\repositories\xmake-repo>git diff

**Reply by free2king** [reply]
检查已经刷出非常长的列表,已经好了，非常感谢！
